### ISSUE TITLE
Policy Gradient Methods: Loss function of policy estimator in REINFORCE

### ISSUE THREAD

**Issue by ArikVoronov** [dennybritz/reinforcement-learning#181]
The tf loss is defined as:

`self.loss = -tf.log(self.picked_action_prob) * self.target`
Where ` self.target ` is the advantage function. 

But as I understand it, in policy gradient methods, that's the actual gradient of parameters, not some loss.  Shouldn't the step just be :
![image](https://user-images.githubusercontent.com/43729836/47279392-6db50700-d5d9-11e8-87dc-58ba3624a106.png)

So why are we minimizing the loss? Is it just a trick since grad= 0 is a minimum point? Why use tf at all?

**Comment by mobil787** [comment]
The goal of loss-function is to find the minimum loss; Policy Gradient Methods assure each iteration step going to the right-way(minimizing loss). If only looking at the gradient=0, it could lead to a local/globe maximum value.

**Comment by ArikVoronov** [comment]
Thanks for the answer, but that's not exactly what I asked, let me clarify:

Why does the code use a tf built-in optimization function and why is the loss defined as the **gradient** of the total policy return? 
Shouldn't we instead analytically derive the explicit gradient as function of state/action and iterate the policy weights  (theta) directly, the way David Silver explains in lecture 7? (the implicit formula in my original question)

**Comment by abhayraw1** [comment]
@ArikVoronov  your intuition is correct. [This lecture](https://web.stanford.edu/class/cs20si/lectures/CS20_intro_to_RL.pdf) on policy gradients looks into this problem in slide 28.
